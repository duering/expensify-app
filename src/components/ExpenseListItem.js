import React from 'react';
import moment from 'moment';
import numeral from 'numeral';
import { Link } from 'react-router-dom'
import { connect } from 'react-redux';
import { removeExpense } from '../actions/expenses';

const ExpenseListItem = ({ id, description, amount, createdAt }) => (
	<div>
		<Link to={`/edit/${id}`}><h3>{description}</h3></Link>
		<p>
		Amount: {numeral(amount).format('$0,0.00')}, 
		Created At: {moment(createdAt).format('MMMM Do, YYYY')}
		</p>
	</div>
);

export default ExpenseListItem;