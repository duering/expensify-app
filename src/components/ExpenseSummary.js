import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import getExpensesTotal from '../selectors/expenses-total';
import selectExpense from '../selectors/expenses';

export const ExpenseSummary = ({ expenseCount, expensesTotal }) => {
	const expenseWord = expenseCount === 1 ? 'expense' : 'expenses';
	const formattedExpensesTotal = numeral(expensesTotal / 100).format('$0,0.00');
	return (
		<h1>Viewing {expenseCount} {expenseWord} totalling {formattedExpensesTotal}</h1>
	);
};

const mapStateToProps = (state) => {
	const visibleExpenses = selectExpense(state.expenses, state.filters);
	return {
		expenseCount: visibleExpenses.length,
		expensesTotal: getExpensesTotal(visibleExpenses)
	};
};

export default connect(mapStateToProps)(ExpenseSummary);