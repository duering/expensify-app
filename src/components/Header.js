import React from 'react';
import { NavLink } from 'react-router-dom';
import { connect } from 'react-redux';
import { startLogout } from '../actions/auth';

export const Header = ({ startLogout }) => (
	<div>
		<h1>Expensify</h1>
		<NavLink to="/" activeClassName="is-active" exact={true}>Dashboard</NavLink><br />
		<NavLink to="/create" activeClassName="is-active">Create Expense</NavLink><br />
		<NavLink to="/help" activeClassName="is-active">Help</NavLink>
		<button onClick={startLogout}>Logout</button>
	</div>
);

const mapDispathToProps = (dispatch) => ({
	startLogout: () => dispatch(startLogout())
});

export default connect(undefined, mapDispathToProps)(Header);