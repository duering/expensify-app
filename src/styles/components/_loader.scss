.loader {
	align-items: center;
	display: flex;
	height: 100vh;
	justify-content: center;
	wright: 100vw;
}

.loader__image {
	height: 6rem;
	weight: 6rem;
}