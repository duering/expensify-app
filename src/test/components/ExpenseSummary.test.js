import React from 'react';
import { shallow } from 'enzyme';
import expenses from '../fixtures/expenses';
import { ExpenseSummary } from '../../components/ExpenseSummary';

test('should correctly render ExpenseSummary with 1 expense', () => {
  const wrapper = shallow(<ExpenseSummary expenseCount={1} expensesTotal={320} />);
  expect(wrapper).toMatchSnapshot();
  expect(wrapper.text()).toEqual('Viewing 1 expense totalling $3.20')
});

test('should correctly render ExpenseSummary with multiple expenses', () => {
  const wrapper = shallow(<ExpenseSummary expenseCount={2} expensesTotal={200} />);
  expect(wrapper).toMatchSnapshot();
  expect(wrapper.text()).toEqual('Viewing 2 expenses totalling $2.00')
});