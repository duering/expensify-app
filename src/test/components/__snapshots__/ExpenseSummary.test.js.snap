// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render ExpenseSummary with 1 expense 1`] = `
<h1>
  Viewing 
  1
   
  expense
   totalling 
  $3.20
</h1>
`;

exports[`should correctly render ExpenseSummary with multiple expenses 1`] = `
<h1>
  Viewing 
  2
   
  expenses
   totalling 
  $2.00
</h1>
`;
